lib/mongoid/history/trackable.rb

```diff
@@ -318,16 +322,25 @@ def next_version
         def increment_current_version
           next_version.tap { |version| send("#{history_trackable_options[:version_field]}=", version) }
         end
+      end
 
+      module MyProtectedInstanceMethods
         protected
 
         def track_history_for_action?(action)
           track_history? && !(action.to_sym == :update && modified_attributes_for_update.blank?)
         end
 
+        def increment_current_version?(action)
+          !(
+            (action == :destroy) ||
+            ancestor_flagged_for_destroy?(_parent)
+          )
```

Comment:
I think this could be clearer and less awkwardly formatted.
```suggestion
          action != :destroy && !ancestor_flagged_for_destroy?(_parent)
```

Reply:
Changed.